pylint/checkers/variables.py

```diff
@@ -217,6 +219,27 @@ def _flattened_scope_names(iterator):
     return set(itertools.chain.from_iterable(values))
 
 
+def _get_for_loop_assign_names(variable):
+    """
+    Get the list of loop variables assigned to by a for loop.
+
+    :param variable: An assigned-to variable. Usually this is the result of
+        `for_node.target`.
+    :type variable: astroid.NodeNG
+
+    :returns: A list of variable names that the for loop assigns to.
+    :rtype: list
+    """
+    variables = []
```

Comment:
Do you have to use recursion here? This can probably be implemented in an iterative way, something as in:

```
    import astroid
    import collections


    def _node_elements(node):
        queue = collections.deque([node])
        while queue:
            elem = queue.popleft()
            if isinstance(elem, astroid.AssignName):
                yield elem.name
            elif isinstance(elem, astroid.Tuple):
                queue.extendleft(elem.get_children())


    p = astroid.parse('for (a, (b, c, (d, e, f))) in i: pass').body[0].target

    for i in _node_elements(p):
        print(i)
            
```